src/render.rs

```diff
@@ -11,7 +11,7 @@ use ratatui::{buffer::Buffer, layout::Rect, widgets::Widget};
 pub fn is_in_bounds<T>(position: &Coordinate, buffer: &Array2<T>) -> bool {
     let (height, width) = buffer.dim(); // Get the dimensions of the buffer
 
-    position.x < width || position.y < height
+    position.x < width && position.y < height
```

Comment:
It still crashes when the board is too but, but I think this is correct and the old version was wack.

Reply:
At least we know that it's probably not because this fn is wrong now.